File: Week4/Training/MVVMArchitectureComp/app/src/main/java/com/example/admin/mvvmarchitecturecomp/data/remote/RemoteService.java
package com.example.admin.mvvmarchitecturecomp.data.remote;

import com.example.admin.mvvmarchitecturecomp.data.remote.models.WeatherApiResponse;

import io.reactivex.Flowable;
import io.reactivex.Single;
import retrofit2.http.GET;
import retrofit2.http.Query;

public interface RemoteService {

    @GET("data/2.5/forecast")
    Single<WeatherApiResponse> getWeatherForecast(@Query("id") String cityId, @Query("APPID") String apiKey);
}

File: Week5/Homework/CodeChallengeKotlin/app/src/main/java/com/example/admin/codechallengekotlin/data/remote/RemoteService.kt
package com.example.admin.codechallengekotlin.data.remote

import com.example.admin.codechallengekotlin.data.remote.models.MetaWeatherResponse
import kotlinx.coroutines.experimental.Deferred
import retrofit2.http.GET
import retrofit2.http.Path

interface RemoteService {
    @GET("api/location/{woeid}/{date}")
    fun getWeatherForecast(@Path("woeid") woeid:String, @Path("date") date:String) : Deferred<List<MetaWeatherResponse>>
}